### PR TITLE
1. Workspace App registration Admin Consent

### LINKED ISSUE #2086: Automation Admin App is not granted Admin Consent to the Workspace
**Describe the bug**
The Automation Admin App is not granted Admin Consent to the Workspace. This means that the E2E tests do not work.

**Steps to reproduce**

1. Run make auth
2. 1. Go to the AAD portal and select the Automation Admin App
3. Select API Permissions
4. You can see that there is an orange warning triangle next to the WorkspaceOwner application permission

### LINKED ISSUE #2110: make auth creates workspace app reg details when auto registration is true
I don't believe `WORKSPACE_API_CLIENT_ID` and secret should be generated by `make auth` when `AUTO_WORKSPACE_APP_REGISTRATION=true`.

### PR DESCRIPTION
**Pull request by ross-p-smith**
# Refactor Auth scripts

## What is being addressed

- Split each Identity into its own script (apart from the API and the UX as they need to be created together).
- Removed the horrible deprecated message that had appeared
- Tidied up the messages in the output
- Created a new function to create Service Principals
- Updated docs
- Fixes #2110 
- Fixes #2086

### CODE REVIEW

**Review by tamirkamara — COMMENTED** [review]
A couple of small comments. I tried to use the new values from the scripts, but got errors using them. Will reach out directly...

File: devops/scripts/aad/create_or_update_service_principal.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+set -e
```
[the remark this answers is not in the document]
Comment (ross-p-smith) [reply devops/scripts/aad/create_or_update_service_principal.sh]: Done

File: devops/scripts/aad/create_application_administrator.sh [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 # AZURE_CORE_OUTPUT=jsonc # force CLI output to JSON for the script (user can still change default for interactive usage in the dev container)
 
 function show_usage()
 {
     cat << USAGE
 
-Utility script for creating an application administrator for TRE. This is optional and is normal
-if you want to delegate Application creation to TRE.
-This script is trigger by the environment variable AUTO_WORKSPACE_APP_REGISTRATION.
+Utility script for creating an application administrator for TRE. This is mandatory and is used
```
Comment (tanya-borisova) [inline devops/scripts/aad/create_application_administrator.sh:11]:
```suggestion
Utility script for creating an application administrator for TRE. It is used
```

File: devops/scripts/aad/create_automation_administrator.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+set -euo pipefail
+# Use this for debug only
+# set -o xtrace
+# AZURE_CORE_OUTPUT=jsonc # force CLI output to JSON for the script (user can still change default for interactive usage in the dev container)
```
Comment (tanya-borisova) [inline devops/scripts/aad/create_automation_administrator.sh:5]: Do you still need this comment?

File: devops/scripts/aad/create_or_update_service_principal.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      sleep 5
+      output=$(az rest --method GET --uri "${msGraphUri}/servicePrincipals/${servicePrincipalId}" 2>/dev/null || true)
+  done
+
+  if [[ -z $output ]]; then
+      echo "Failed"
+      exit 1
+  fi
+
+  #echo "Service principal with ID ${servicePrincipalId} found"
```
Comment (tanya-borisova) [inline devops/scripts/aad/create_or_update_service_principal.sh:29]: Nit: remove this comment (or uncomment?)

File: devops/scripts/aad/create_workspace_application.sh [hunk 5]
```diff
@@ -0,0 +1,374 @@
+#!/bin/bash
+set -euo pipefail
+# Use this for debug only
+# set -o xtrace
+# AZURE_CORE_OUTPUT=jsonc # force CLI output to JSON for the script (user can still change default for interactive usage in the dev container)
```
Comment (tanya-borisova) [inline devops/scripts/aad/create_workspace_application.sh:5]: Nit: do you still need this comment?

**Review by tanya-borisova — APPROVED** [review]
Some small nits but no concerns. I've ran it also on my environment and things seem to work 👍

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**